test.js

```diff
@@ -482,6 +482,35 @@ test('keep alive', function (t) {
   }
 })
 
+test('keep-alive messages are not forwarded to the noise stream', async function (t) {
+  t.plan(2)
+
+  const a = new NoiseStream(true)
+  const b = new NoiseStream(false)
+
+  a.setKeepAlive(1)
+  // No keep-alive for b
+
+  a.resume()
+
+  a.rawStream.pipe(b.rawStream).pipe(a.rawStream)
+
+  b.on('data', function (data) {
+    if (data.toString() !== 'an actual message') {
+      t.fail('keep-alive message forwarded to the stream')
+      return
+    }
+    t.pass('Received the actual message')
+    t.is(rawStreamCounter > 5, true, 'sanity check: keep-alive packets were being sent')
+  })
+
+  // TODO: figure out why listening to b.rawStream doesn't work
```

Comment:
Any idea why `b.rawStream.on('data')` is not yielding, but `_rawStream` is?
It's not fundamentally important for this test, I can also remove this sanity check, but I found it odd. It seems to be related to this code path: https://github.com/holepunchto/hyperswarm-secret-stream/blob/7f6737be68644aed2f131e719b4900db5327d5b8/index.js#L111-L114, but I didn't fully trace it

Reply:
I wanted to add a sanity check to the test, to verify that the raw stream is receiving the keep-alive packets. That way, I'm sure that the NoiseStream is indeed filtering them out. This avoids a false positive if no keep-alive packets are sent.

I tried adding a listener to `b.rawStream.on('data')`, but it is not triggering, even though I verified with `console.log()` that the packets are arriving.
Adding a listener like `b._rawStream.on('data')` does trigger every time a keep-alive packet arrives.

I didn't really understand the difference between `_rawStream` and `rawStream`, so I'm not sure why they behave differently here. The other tests all use `rawStream`, so I might be messing something up.

Reply:
ah ya they are reversed, writing to one makes the other emit data and vice versa